src/main/java/subway/service/StationPathService.java

```diff
@@ -0,0 +1,97 @@
+package subway.service;
+
+import lombok.RequiredArgsConstructor;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import subway.domain.Station;
+import subway.domain.StationLine;
+import subway.domain.StationLineRepository;
+import subway.domain.StationRepository;
+import subway.exception.StationLineSearchFailException;
+import subway.service.dto.StationPathResponse;
+import subway.service.dto.StationResponse;
+
+import java.math.BigDecimal;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class StationPathService {
+    private final StationLineRepository stationLineRepository;
+    private final StationRepository stationRepository;
+
+    @Transactional(readOnly = true)
+    public StationPathResponse searchStationPath(Long startStationId, Long destinationStationId) {
+        checkValidPathFindRequest(startStationId, destinationStationId);
+
+        final List<Station> totalStations = stationRepository.findAll();
+
+        final GraphPath<Long, DefaultWeightedEdge> path = calculatePath(startStationId, destinationStationId, totalStations);
```

Comment:
경로 조회를 위한 로직이 모두 Service에 있네요
만약 경로 조회에 사용하는 라이브러리를 변경하거나 길찾기에 적용되는 다양한 전략(최소 환승, 최소 요금 등)이 적용되어야 한다면 서비스 코드의 변경량이 많아질 것 같아요 
변경을 최소화하기 위해 서비스에 로직의 로직을 도메인 객체를 추가하면 어떨까요?
```suggestion
        final Path path = new Path(startStationId, destinationStationId, totalStations);
```

Reply:
@testrace 경로 조회의 로직이 요구사항에 따라 변화에 민감할것 같다는 부분에 저도 굉장히 공감되네요
제안해주신 도메인의 로직으로 서비스의 로직을 뺀다는것은 StationLine 내부로 해당 로직을 옮기는게 좋다는 의견일까요?
경로를 계산하는데에 전체 StationLine 목록과 전체 Station 목록 데이터를 가지고 있어야해서 따로 도메인 서비스로 빼고 인터페이스에 서비스가 의존하도록 해서 변화에 취약해지지 않도록 하는 방법도 있을것 같은데 의견 부탁드립니다. 더 자세히 논의해보고 싶으면 슬렉의로 논의를 해봐도 좋을것같습니다 ㅎㅎ